pkg/gui/controllers/helpers/refs_helper.go

```diff
@@ -271,6 +272,57 @@ func (self *RefsHelper) CreateGitResetMenu(ref string) error {
 	})
 }
 
+func (self *RefsHelper) CreateCheckoutMenu(commit *models.Commit) error {
+	branches := lo.Filter(self.c.Model().Branches, func(branch *models.Branch, _ int) bool {
+		return commit.Hash == branch.CommitHash && branch.Name != self.c.Model().CheckedOutBranch
+	})
+
+	ref := commit.Hash
+	var menuItems []*types.MenuItem
+
+	menuItems = append(menuItems, &types.MenuItem{
+		LabelColumns: []string{fmt.Sprintf("%s %s", self.c.Tr.CheckoutCommit, self.c.Tr.AsDetachedHead), style.FgRed.Sprint(ref[:8])},
```

Comment:
As a general rule of thumb, you should never concatenate or otherwise assemble several translated strings. You never know, it might not work well in other languages (order might have to be different). It is better to introduce a new text that is the assembled sentence you need (`CheckoutAsDetachedHead` in this case, with the text "Checkout commit as detached head").